### PR TITLE
show animated gifs on public sharing page

### LINKED ISSUE #13164: Animated gifs do not animate
Hello,

I recently installed OwnCloud 7.0.4 and discovered that animated gifs in the slideshow, gallery, or single file share modes do not animate.
From what I recall, animated gifs did animate in owncloud 6.x.

[Here is a link to an example image, an animated gif of a cat playing with and popping ballons on my install.](https://cloud.xfcab.com/public.php?service=files&t=5c6e6b07bae06e79fefbc874e4a3f70c)

### PR DESCRIPTION
**Pull request by georgehrke**
fixes #13164

### CODE REVIEW

File: config/config.sample.php [hunk 1]
```diff
@@ -944,6 +944,15 @@
 'forwarded_for_headers' => array('HTTP_X_FORWARDED', 'HTTP_FORWARDED_FOR'),
 
 /**
+ * max file size for animating gifs on public-sharing-site.
+ * If the gif is bigger, it'll show a static preview
+ *
+ * Default is 10MB
+ * Set to -1 for no limit
+ */
+'max_filesize_animated_gifs_public_sharing' => 1024 * 1024 * 10,
```
Comment (LukasReschke) [inline config/config.sample.php]: That notation can be kinda confusing? – Can we not just use a simple integer in MB values? – Nobody will ever have the need to define this down to a single specific byte.

Also make clear whether this are mebibyte or megabyte.

File: apps/files_sharing/templates/public.php [hunk 2]
```diff
@@ -21,6 +21,7 @@
 
 $thumbSize=1024;
 $previewSupported = OC\Preview::isMimeSupported($_['mimetype']) ? 'true' : 'false';
+$maxSizeAnimateGif = OC::$server->getConfig()->getSystemValue('max_filesize_animated_gifs_public_sharing', 1024 * 1024 * 10);
```
Comment (LukasReschke) [inline apps/files_sharing/templates/public.php]: Put that logic in the controller?

File: config/config.sample.php (hunk 1 above)
Comment (georgehrke) [inline config/config.sample.php]: ok, will fix

**Comment by MorrisJobke** [comment]
Works :+1:

**Comment by LukasReschke** [comment]
@owncloud-bot Retest this please

**Comment by LukasReschke** [comment]
@owncloud-bot Retest this please

**Comment by MorrisJobke** [comment]
I will fix that ;)
```
00:14:54.916 There was 1 failure:
00:14:54.916 
00:14:54.916 1) OCA\Files_Sharing\Controllers\ShareControllerTest::testShowShare
00:14:54.920 Failed asserting that two objects are equal.
00:14:54.920 --- Expected
00:14:54.920 +++ Actual
00:14:54.920 @@ @@
00:14:54.920          'fileSize' => '33 B'
00:14:54.920 +        'nonHumanFileSize' => 33
00:14:54.920 +        'maxSizeAnimateGif' => 10
00:14:54.920      )
00:14:54.920      'renderAs' => 'base'
00:14:54.920      'appName' => 'files_sharing'
00:14:54.920      'headers' => Array (...)
00:14:54.920      'cookies' => Array ()
00:14:54.920      'status' => 200
00:14:54.920      'lastModified' => null
00:14:54.920      'ETag' => null
00:14:54.920  )
```

**Comment by MorrisJobke** [comment]
Fixed

**Comment by MorrisJobke** [comment]
@owncloud-bot retest this please

**Comment by MorrisJobke** [comment]
@owncloud-bot retest this please

File: apps/files_sharing/js/public.js [hunk 3]
```diff
@@ -97,7 +97,14 @@ OCA.Sharing.PublicApp = {
 		};
 
 		var img = $('<img class="publicpreview" alt="">');
-		if (previewSupported === 'true' || mimetype.substr(0, mimetype.indexOf('/')) === 'image' && mimetype !== 'image/svg+xml') {
+
+		var filesize = parseInt($('#filesize').val());
```
Comment (DeepDiver1975) [inline apps/files_sharing/js/public.js]: radix is missin ong parseInt

File: apps/files_sharing/js/public.js [hunk 4]
```diff
@@ -97,7 +97,14 @@ OCA.Sharing.PublicApp = {
...
+		var max_animated_gif_size = parseInt($('#maxSizeAnimateGif').val());
+
+		if (mimetype === 'image/gif' && (max_animated_gif_size === -1 || filesize <= (max_animated_gif_size * 1024 * 1024))) {
```
Comment (DeepDiver1975) [inline apps/files_sharing/js/public.js]: line too long

File: apps/files_sharing/js/public.js (hunk 3 above)
Comment (DeepDiver1975) [inline apps/files_sharing/js/public.js]: filesize => fileSize

File: apps/files_sharing/js/public.js [hunk 5]
```diff
@@ -97,7 +97,14 @@ OCA.Sharing.PublicApp = {
 		};
 
 		var img = $('<img class="publicpreview" alt="">');
-		if (previewSupported === 'true' || mimetype.substr(0, mimetype.indexOf('/')) === 'image' && mimetype !== 'image/svg+xml') {
+
+		var filesize = parseInt($('#filesize').val());
+		var max_animated_gif_size = parseInt($('#maxSizeAnimateGif').val());
```
Comment (DeepDiver1975) [inline apps/files_sharing/js/public.js]: :camel: case

File: apps/files_sharing/js/public.js [hunk 6]
```diff
@@ -97,7 +97,14 @@ OCA.Sharing.PublicApp = {
...
+
+		if (mimetype === 'image/gif' && (max_animated_gif_size === -1 || filesize <= (max_animated_gif_size * 1024 * 1024))) {
+			img.attr('src', $('#downloadURL').val());
+			img.appendTo('#imgframe');
+		} else if (previewSupported === 'true' || mimetype.substr(0, mimetype.indexOf('/')) === 'image' && mimetype !== 'image/svg+xml') {
```
Comment (DeepDiver1975) [inline apps/files_sharing/js/public.js]: line too long

**Comment by DeepDiver1975** [comment]
please fix scrutinizer issue: https://scrutinizer-ci.com/g/owncloud/core/inspections/35f0333d-f80f-4efd-a787-56a1364cf763/issues/

**Comment by DeepDiver1975** [comment]
besides comments above - tested :+1:

**Comment by georgehrke** [comment]
> $shareTmpl['nonHumanFileSize'] is of type false|string|null, but the function expects a integer.

\OC\Files\Filesystem::filesize has no phpdoc block and the error message seems like some strange error in scrutinizer

**Comment by DeepDiver1975** [comment]
> \OC\Files\Filesystem::filesize has no phpdoc block and the error message seems like some strange error in scrutinizer

```php
		$nonHumanFileSize = \OC\Files\Filesystem::filesize($originalSharePath);
		$shareTmpl['nonHumanFileSize'] = $nonHumanFileSize;
		$shareTmpl['fileSize'] = \OCP\Util::humanFileSize($nonHumanFileSize);

**Comment by MorrisJobke** [comment]
@owncloud-bot retest this please

**Comment by georgehrke** [comment]
mimimi, I pulled, why did it overwrite morris' changes ...

**Comment by MorrisJobke** [comment]
> mimimi, I pulled, why did it overwrite morris' changes ...

You need to delete the branch and create a new one with the origin as source ;)

**Comment by georgehrke** [comment]
@owncloud-bot retest this please

**Comment by georgehrke** [comment]
@owncloud-bot retest this please

**Comment by MorrisJobke** [comment]
Still works :+1:

**Comment by DeepDiver1975** [comment]
THX @georgehrke

**Pull request merged**